content/blog/2025/06/2025-06-13-twim.md

```diff
@@ -0,0 +1,225 @@
+> * [#5210](https://github.com/matrix-org/matrix-rust-sdk/pull/5210) Multiverse supports threads!
+> * [#5198](https://github.com/matrix-org/matrix-rust-sdk/pull/5198) When a client doesn't support threads, threads are rendered as replies: each message is marked as a reply of the previous message in the same thread. What if your client supports threads and one thread contains a real reply to a message? Well, it wasn't working well. Now it works well.
+> * [#5185](https://github.com/matrix-org/matrix-rust-sdk/pull/5185) The `Timeline` gains superpowers to filtering out messages that are part of a thread, so they can be rendered separately.
+> 
+> Threads are making great progresses, but events are not persisted in a storage so far. Happily, progress on that front has been made too!
+> 
+> * [#5172](https://github.com/matrix-org/matrix-rust-sdk/pull/5172) and [#5182](https://github.com/matrix-org/matrix-rust-sdk/pull/5182) prepares the event cache for storing a linked chunk per thread.
+> 
+> #### WebAssembly
+> 
+> Work on WebAssembly pursues. This time, the first set of patches to make the `EventCacheStore` compatible with [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) has been merged.
+> 
+> * [#5138](https://github.com/matrix-org/matrix-rust-sdk/pull/5138) The initial types and the database migrations for `EventCacheStore` on IndexedDb has landed. It's the foundation for the next patches to make the whole `EventCacheStore` compatible with IndexedDB when compiled to `wasm32-unknown-unknown`.
+> 
+> #### Crypto
+> 
+> * [#5219](https://github.com/matrix-org/matrix-rust-sdk/pull/5219) If the SDK believes that the sender of a room event has been spoofed, the SDK will use a dedicated verification level, possibly allowing us to treat such events more carefully. This new verification level is `MismatchedSender`.
+> 
+> #### General
+> 
+> * [#5205](https://github.com/matrix-org/matrix-rust-sdk/pull/5205) We are reverting our experiment around `busy_timeout` with SQLite to compare the data and see if it's helping fixing some lock issues.
+> * [#5099](https://github.com/matrix-org/matrix-rust-sdk/pull/5099) When using `Client::add_event_handler`, it is now possible to define a callback with an argument of kind `Option<EncryptionInfo>`, which is particularly useful and requested by the community frequently!
+
+### Matrix Dart SDK ([website](https://github.com/famedly/matrix-dart-sdk))
+
+Matrix SDK written in pure Dart.
+
+[td](https://matrix.to/#/@td:technodisaster.com) reports
+
+> #### Released v1.0.0 this week
+> 
+> * This major new release now move away from libolm to vodozemac. Fluffychat now has a call for testing open here https://github.com/krille-chan/fluffychat/issues/1957
+> * v1.14 spec endpoints support
+> * A bunch of other breaking changes since the last announcement, https://github.com/famedly/matrix-dart-sdk/releases
```

Comment:
```suggestion
> * This major new release now move away from libolm to vodozemac. Fluffychat now has a call for testing open here <https://github.com/krille-chan/fluffychat/issues/1957>
> * v1.14 spec endpoints support
> * A bunch of other breaking changes since the last announcement, <https://github.com/famedly/matrix-dart-sdk/releases>
```